src/index.ts

```diff
@@ -145,14 +145,14 @@ export default class Header {
    * @private
    */
   normalizeData(data: HeaderData): HeaderData {
-    const newData: HeaderData = {text: '', level: this.defaultLevel.number };
+    const newData: HeaderData = { text: '', level: this.defaultLevel.number };
 
     if (typeof data !== 'object') {
-      data = {text: '', level: this.defaultLevel.number};
+      data = { text: '', level: this.defaultLevel.number};
     }
 
     newData.text = data.text || '';
-    newData.level = parseInt(data.level.toString()) || this.defaultLevel.number;
+    newData.level = data.level ? parseInt(data.level.toString()) : this.defaultLevel.number;
```

Comment:
how level can be undefined here?

Reply:
probably with the casting of `HeaderData`

Reply:
maybe `typeof data !== 'object'` is not working as expected? Have you managed to reproduce the problem? Try adding a Header via Toolbox and check what is the `data` exactly?

Reply:
I tested it locally, it seems to be working fine I guess

Reply:
yes, but the reason is still not clear

Reply:
I mean there is an issue with types somewhere

Reply:
seems like in `ConstructorArgs` the data should be `HeaderData | {}`

Reply:
I guess if `HeaderData | {}` is used instead of `HeaderData` for data, we will need to check for availability of level and text inside the object.

Reply:
After debugging, it seems that the data received in `normalizeData` function is an empty object. Before using header tool, the data is an empty object. Until initializing and using header tool, by filling the text, then the object get filled with its appropriate infos.

Reply:
yes, statement is ok, but we need to improve types as well

Reply:
and also `if (typeof data !== 'object') {` will never fired

Reply:
no it will not. I guess should check data content instead

Reply:
you can use a type guard util like this

```ts
isHeaderData(data: any): data is HeaderData {
  return (data as HeaderData).text !== undefined;
}
```